src/gc.c

```diff
@@ -1670,10 +1711,48 @@ static void gc_sweep_pool(int sweep_full)
             p->newpages = NULL;
         }
     }
+        // wait for concurrent sweep to finish
+    if (jl_n_gcthreads != 0) {
+        while (jl_atomic_load(&gc_concurrent_sweep_running)) {
```

Comment:
So this would mean that the concurrent sweep overlapped with the mark phase?

Reply:
It's possible that part of a sweep from the (N-1)th GC overlaps with the mark-phase of the Nth GC.

I believe this should be safe since the concurrent portion of sweeping is just returning pages to `pagetable_t memory_map`.